src/containers/Passphrase.jsx

```diff
@@ -0,0 +1,64 @@
+import { connect } from 'react-redux'
+import get from 'lodash/get'
+import compose from 'lodash/flowRight'
+import Alerter from 'cozy-ui/react/Alerter'
+import { translate } from 'cozy-ui/react/I18n'
+import { AUTH_MODE } from 'actions/twoFactor'
+import {
+  updatePassphrase,
+  updatePassphrase2FAFirst,
+  updatePassphrase2FASecond
+} from 'actions/passphrase'
+import { fetchInfos } from 'actions'
+import PassphraseView from 'components/PassphraseView'
+
+const mapStateToProps = state => ({
+  fields: state.fields,
+  isTwoFactorEnabled:
+    get(state, 'fields.auth_mode.value') === AUTH_MODE.TWO_FA_MAIL,
+  instance: state.instance,
+  passphrase: state.passphrase
+})
+
+const mapDispatchToProps = (dispatch, ownProps) => ({
+  onPassphraseSimpleSubmit: (current, newVal) => {
+    return (
+      dispatch(updatePassphrase(current, newVal))
+        .then(() => Alerter.info(ownProps.t('ProfileView.password.reload')))
+        // eslint-disable-next-line no-console
+        .catch(e => console.error(e))
+    )
+  },
+  onPassphrase2FAStep1: current => {
+    return dispatch(updatePassphrase2FAFirst(current)).catch(e =>
+      // eslint-disable-next-line no-console
+      console.error(e)
+    )
+  },
+  onPassphrase2FAStep2: (current, newVal, twoFactorCode, twoFactorToken) => {
+    return (
+      dispatch(
+        updatePassphrase2FASecond(
+          current,
+          newVal,
+          twoFactorCode,
+          twoFactorToken
+        )
+      )
+        .then(() => Alerter.info(ownProps.t('ProfileView.password.reload')))
+        // eslint-disable-next-line no-console
+        .catch(e => console.error(e))
+    )
```

Comment:
Is there a reason not to use await/async ?

Reply:
Nothing more than "I copy/pasted it from another file and left it as is". It's totally possible to use await/async

Reply:
See https://github.com/cozy/cozy-settings/pull/269/commits/2f531491dd22f5685ff995fbe3d0a404b3c51633